microsoft/terminal
# Define Automation Properties for TermControl

## Summary of the Pull Request
Defines the following automation properties for a Terminal Control:
- [**Orientation**](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.getorientationcore):
  - The orientation of the control
  - `None` --> `Vertical`
- [**Name**](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.getnamecore):
  - The name as used by assistive technology and other Microsoft UI Automation clients. Generally presented by automation clients as the primary way to identify an element (along with the control type)
  - `""` --> `<profile name>`
- [**HelpText**](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.gethelptextcore):
  - The help text. Generally presented by automation clients if requested by the user. This would be something that you would normally expect to appear from tooltips.
  - `""` --> `<tab title>`
- [**LiveSetting**](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.getlivesettingcore):
  - reports the live setting notification behavior. A representation of how assertive this control should be when content changes.
  - `none` --> `Polite`

## References
#2099 - Automation Properties for TerminalControl, Search Box
#2142 - Localization

## PR Checklist
* [X] Closes #2142

## Detailed Description of the Pull Request / Additional comments
`ProfileName` had to be added to the TerminalSettings (IControlSettings) to pass that information along to the automation peer. In the rare event that somebody purposefully decided to make their `ProfileName` empty, we fallback to the tab title.


## Validation Steps Performed
Verified using Accessibility Insights and inspect.exe

This is are some examples of the information a general user can expect to receive about a Terminal Control.
```
- Type: Terminal Control
- Name: Command Prompt
- Help Text (if requested): Command Prompt - ping bing.com
```
```
- Type: Terminal Control
- Name: Ubuntu
- Help Text (if requested): cazamor@PC-cazamor:/mnt/c/Users/cazamor$
```
Note, it is generally read by an automation client as follows:
`"<type>, <name>"`
